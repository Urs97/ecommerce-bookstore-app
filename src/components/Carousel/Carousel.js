import React from 'react';
import { CarouselProvider, Slider, Slide, Image, DotGroup, ButtonBack, ButtonNext } from 'pure-react-carousel';
import 'pure-react-carousel/dist/react-carousel.es.css';
import './Carousel.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleLeft } from '@fortawesome/free-solid-svg-icons';
import { faAngleRight } from '@fortawesome/free-solid-svg-icons';
import SlideContent from './SlideContent/SlideContent';
import photo_1_sm from '../../assets/images/lib-photo-1-sm.jpg';
import photo_1_md from '../../assets/images/lib-photo-1-md.jpg';
import photo_1_lg from '../../assets/images/lib-photo-1-lg.jpg';
import photo_2_sm from '../../assets/images/lib-photo-2-sm.jpg';
import photo_2_md from '../../assets/images/lib-photo-2-md.jpg';
import photo_2_lg from '../../assets/images/lib-photo-2-lg.jpg';
import photo_3_sm from '../../assets/images/lib-photo-3-sm.jpg';
import photo_3_md from '../../assets/images/lib-photo-3-md.jpg';
import photo_3_lg from '../../assets/images/lib-photo-3-lg.jpg';

class Carousel extends React.Component {
    render() {
        const text_1 = "Reading provides you with wisdom";
        const text_2 = "Learn from the experience of others";
        const text_3 = "Read and relax after a long day";
        const title_1 = "Keep Reading";
        const title_2 = "Knowledge Is Power";
        const title_3 = "Reading Is Fun";

        return (
            <CarouselProvider
                className="carousel-provider"
                naturalSlideWidth={16}
                naturalSlideHeight={9}
                totalSlides={3}
                isPlaying={true}
                infinite={true}
                hasMasterSpinner={true}>
                <Slider className="slider">
                    <Slide index={0} className="slide">
                        <SlideContent 
                            text={text_1}
                            title={title_1}
                        />
                        <Image src={photo_1_sm}
                            srcSet={`${photo_1_sm} 640w,
                                    ${photo_1_md} 1024w,
                                    ${photo_1_lg} 2400w,`} 
                            alt="beautiful library interior"/>
                    </Slide>
                    <Slide index={1} className="slide">
                        <SlideContent 
                            text={text_2}
                            title={title_2}
                        />
                        <Image src={photo_2_sm}
                            srcSet={`${photo_2_sm} 640w,
                                    ${photo_2_md} 1024w,
                                    ${photo_2_lg} 2400w,`} 
                            alt="beautiful library interior"/>
                    </Slide>
                    <Slide index={2} className="slide">
                        <SlideContent 
                            text={text_3}
                            title={title_3}
                        />
                        <Image src={photo_3_sm}
                            srcSet={`${photo_3_sm} 640w,
                                    ${photo_3_md} 1024w,
                                    ${photo_3_lg} 2400w,`} 
                            alt="beautiful library interior"/>
                    </Slide>
                </Slider>
                <ButtonBack className="carousel-btn">
                    <span role="img" aria-label="left arrow icon">
                        <FontAwesomeIcon icon={faAngleLeft} />
                    </span>
                </ButtonBack>
                <ButtonNext className="carousel-btn crs-btn-next">
                    <span role="img" aria-label="right arrow icon">
                        <FontAwesomeIcon icon={faAngleRight} />
                    </span>
                </ButtonNext>
                <DotGroup className="dot-grp"/>
            </CarouselProvider>
        )
    };
};

export default Carousel;