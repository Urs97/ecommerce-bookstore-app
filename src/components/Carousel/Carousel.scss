@import '../../styles/shared.scss';

.carousel-provider {
    width: 100%;
    height: 80vh;
    position: relative;

    .slider {
        width: 100%;
        height: 100%;

        .slide {
            position: relative;
        }
    
        .slide::after {
            content: '';
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            z-index: 1;
            background-color: rgba(0, 0, 0, 0.5);
        }
    }

    .carousel-btn {
        display: none;
        position: absolute;
        top: 50%;
        left: 0;
        -webkit-transform: translateY(-50%);
        -ms-transform: translateY(-50%);
        transform: translateY(-50%);
        background-color: rgba(0, 0, 0, 0.7);
        outline: none;
        border: none;
        text-decoration: none;
        padding: 1.75em 0.75em;

        span svg {
            color: #FFFFFF;
            font-size: 1.25em;
        }
    }

    .carousel-btn:hover {
        background-color: rgba(0, 0, 0, 0.95);
    }

    .carousel-btn.crs-btn-next {
        right: 0;
        left: auto;
    }

    .dot-grp {
        position: absolute;
        z-index: 2;
        bottom: 1.5em;
        left: 0;
        right: 0;
        margin-left: auto;
        margin-right: auto;
        width: 10em;
        display: flex;
        justify-content: center;
        
        & button {
            padding: 0.5em;
            border-radius: 50%;
            border-color: #000000;
            border-width: 0;
            outline: none;
        }

        & button:focus {
            background-color: #A9A9A9;
        }

        & .carousel__dot--selected {
            background-color: $blue;
        }

        & button:not(:last-child) {
            margin-right: 0.2em;
        }

    }
}

.carousel-provider:hover .carousel-btn {
    display: inline-block;
}

