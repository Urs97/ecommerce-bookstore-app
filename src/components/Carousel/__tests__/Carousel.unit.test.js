import React from "react";
import { shallow } from "enzyme";
import Carousel from "../Carousel";

describe("Carousel", () => {
  test("renders Carousel", () => {
    const wrapper = shallow(<Carousel />);
    expect(wrapper).toMatchSnapshot();
  });
});
