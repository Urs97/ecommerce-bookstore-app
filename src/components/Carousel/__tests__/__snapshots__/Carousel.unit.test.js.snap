// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders Carousel 1`] = `
<n
  className="carousel-provider"
  currentSlide={0}
  disableAnimation={false}
  disableKeyboard={false}
  dragEnabled={true}
  dragStep={1}
  hasMasterSpinner={true}
  infinite={true}
  interval={5000}
  isIntrinsicHeight={false}
  isPageScrollLocked={false}
  isPlaying={true}
  lockOnWindowScroll={false}
  naturalSlideHeight={9}
  naturalSlideWidth={16}
  orientation="horizontal"
  playDirection="forward"
  step={1}
  tag="div"
  totalSlides={3}
  touchEnabled={true}
  visibleSlides={1}
>
  <n
    className="slider"
  >
    <n
      className="slide"
      index={0}
    >
      <SlideContent
        text="Reading provides you with wisdom"
        title="Keep Reading"
      />
      <n
        alt="beautiful library interior"
        src="lib-photo-1-sm.jpg"
        srcSet="lib-photo-1-sm.jpg 640w,
                                    lib-photo-1-md.jpg 1024w,
                                    lib-photo-1-lg.jpg 2400w,"
      />
    </n>
    <n
      className="slide"
      index={1}
    >
      <SlideContent
        text="Learn from the experience of others"
        title="Knowledge Is Power"
      />
      <n
        alt="beautiful library interior"
        src="lib-photo-2-sm.jpg"
        srcSet="lib-photo-2-sm.jpg 640w,
                                    lib-photo-2-md.jpg 1024w,
                                    lib-photo-2-lg.jpg 2400w,"
      />
    </n>
    <n
      className="slide"
      index={2}
    >
      <SlideContent
        text="Read and relax after a long day"
        title="Reading Is Fun"
      />
      <n
        alt="beautiful library interior"
        src="lib-photo-3-sm.jpg"
        srcSet="lib-photo-3-sm.jpg 640w,
                                    lib-photo-3-md.jpg 1024w,
                                    lib-photo-3-lg.jpg 2400w,"
      />
    </n>
  </n>
  <n
    className="carousel-btn"
  >
    <span
      aria-label="left arrow icon"
      role="img"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              256,
              512,
              Array [],
              "f104",
              "M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z",
            ],
            "iconName": "angle-left",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </span>
  </n>
  <n
    className="carousel-btn crs-btn-next"
  >
    <span
      aria-label="right arrow icon"
      role="img"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              256,
              512,
              Array [],
              "f105",
              "M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z",
            ],
            "iconName": "angle-right",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </span>
  </n>
  <n
    className="dot-grp"
  />
</n>
`;
